scripts/ai/AIDriveStrategyCourse.lua

```diff
@@ -128,9 +128,31 @@ function AIDriveStrategyCourse:delete()
     AIDriveStrategyCourse:superClass().delete(self)
 end
 
-
 function AIDriveStrategyCourse:getGeneratedCourse(jobParameters)
-    return self.vehicle:getFieldWorkCourse()
+    local course = self.vehicle:getFieldWorkCourse()

```

Comment:
A simple nil check, if a course was found might be needed for the bale finder.

Reply:
Good catch.